code/modules/mob/living/living.dm

```diff
@@ -383,14 +383,18 @@
 
 /mob/living/verb/succumb(whispered as null)
 	set hidden = TRUE
-	if (InCritical() && !HAS_TRAIT(src, TRAIT_NODEATH))
+	if (can_succumb())
 		log_message("Has [whispered ? "whispered his final words" : "succumbed to death"] while in [InFullCritical() ? "hard":"soft"] critical with [round(health, 0.1)] points of health!", LOG_ATTACK)
 		adjustOxyLoss(health - HEALTH_THRESHOLD_DEAD)
 		updatehealth()
 		if(!whispered)
 			to_chat(src, "<span class='notice'>You have given up life and succumbed to death.</span>")
 		death()
 
+/// Returns whether or not the mob can succumb
+/mob/living/proc/can_succumb()
+	return InCritical() && !HAS_TRAIT(src, TRAIT_NODEATH)
```

Comment:
```suggestion
#define CAN_SUCCUMB(target) (target.InCritical() && !HAS_TRAIT(target, TRAIT_NODEATH))
```
Just so we don't tempt future coders to redefine this proc, and instead make their changes more generically.